docs/source/starter/style_guide.rst

```diff
@@ -176,29 +189,33 @@ However, when using DataParallel, you will need to call forward manually
 ****
 Data
 ****
+
 These are best practices for handling data.
 
 Dataloaders
 ===========
-Lightning uses dataloaders to handle all the data flow through the system. Whenever you structure dataloaders,
+
+Lightning uses :class:`~torch.utils.data.DataLoader` to handle all the data flow through the system. Whenever you structure dataloaders,
 make sure to tune the number of workers for maximum efficiency.
 
-.. warning:: Make sure not to use ddp_spawn with num_workers > 0 or you will bottleneck your code.
+.. warning:: Make sure not to use ``Trainer(strategy="ddp_spawn")`` with ``num_workers>0`` in a DataLoader or you will bottleneck your code.
 
 DataModules
 ===========
-Lightning introduced datamodules. The problem with dataloaders is that sharing full datasets is often still challenging
-because all these questions need to be answered:
 
-* What splits were used?
-* How many samples does this dataset have?
-* What transforms were used?
-* etc...
+The :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is designed as a way of decoupling data-related
+hooks from the :class:`~pytorch_lightning.core.lightning.LightningModule` so you can develop dataset agnostic models. It makes it easy to hot swap different
+datasets with your model, so you can test it and benchmark it across domains. It also makes sharing and reusing the exact data splits and transforms across projects possible.
+
+Checkout :ref:`data` document to understand data management within Lightning and its best practices.
```

Comment:
```suggestion
Check out :ref:`data` document to understand data management within Lightning and its best practices.
```
Quiz:
Which of the following is a valid git command?

A) git check out
B) git checkout
C) git chickenout

Reply:
C) :smile: 

oh! man. I swear I checked it on all PRs when you suggested it on one of them :p
I'll include this as a check for the final doc revamp PR.

Reply:
No worries. I'm just making a little fun of git 😄